pipeline_dp/pipeline_backend.py

```diff
@@ -659,11 +659,28 @@ def singleton(self):
 class LocalBackend(PipelineBackend):
     """Local Pipeline adapter."""
 
+    def __init__(self, output_reiterable: bool = True):
```

Comment:
Do we need this flag in general? Should it be set to false by default for backwards compatibility (afaik it is part of public API)? 16% is significant and two times more memory as well, but I think if local is used only for testing then it is still ok.
One big concern: currently LocalBackend is created by the user but we use it internally therefore we will have to handle non-reiterable cases anyway and code with that in mind, so introducing this flag but not enforcing it to True does not help, or?

Reply:
Agree. I've dropped the flag and make it Reiterable always